lldb/unittests/Expression/DWARFExpressionTest.cpp

```diff
@@ -768,3 +768,63 @@ TEST(DWARFExpression, ExtensionsDWO) {
 
   testExpressionVendorExtensions(dwo_module_sp, *dwo_dwarf_unit);
 }
+
+TEST_F(DWARFExpressionMockProcessTest, DW_OP_piece_file_addr) {
+  struct MockTarget : Target {
```

Comment:
You could consider using gmock with [this kind](https://github.com/llvm/llvm-project/blob/main/lldb/unittests/TestingSupport/Host/NativeProcessTestUtils.h#L73) of a pattern that would enable you to write `EXPECT_CALL(target, ReadMemory(0x40, 1)).WillOnce(Return(ByMove(std::vector<uint8_t>{0x11})));`

Reply:
Very cool, thanks for the suggestion. I had to tweak it a little bit since this is using an out parameter for the bugger. I couldn't find a declarative way to do that, but if you know of one I'd love to learn more about it.

Reply:
The code I linked to ([this one](https://github.com/llvm/llvm-project/blob/main/lldb/unittests/TestingSupport/Host/NativeProcessTestUtils.h#L73)) shows one way to do it. Basically, you implement the real function in terms on some other interface -- one which is easier to mock -- and then you mock *that*.